pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionConfigUtils.java

```diff
@@ -367,6 +367,10 @@ public static void inferMissingArguments(FunctionConfig functionConfig) {
         if (functionConfig.getParallelism() == null) {
             functionConfig.setParallelism(1);
         }
+        
+        if (functionConfig.getMaxPendingAsyncRequests() == null) {
+        	functionConfig.setMaxPendingAsyncRequests(Integer.valueOf(1000));
+        }
```

Comment:
Can you have a static default value inside the FunctionConfig itself and use that to init here

Reply:
Also can you do the same for  forwardSourceMessageProperty as well

Reply:
Done